Assets/LeapMotion/Core/Scripts/DataStructures/DeltaBuffer.cs

```diff
@@ -8,108 +8,173 @@
+      return _buffer.Get(idx).value;
+    }
 
-  public void SetLatest(T sample, float sampleTime) {
-    if (Length == 0) Set(0, sample, sampleTime);
-    else Set(Length - 1, sample, sampleTime);
-  }
+    public SampleType GetLatest() {
+      return Get(Count - 1);
+    }
 
-  public float GetTime(int idx) {
-    return _buffer.Get(idx).time;
-  }
+    public void Set(int idx, SampleType sample, float sampleTime) {
+      _buffer.Set(idx, new ValueTimePair { value = sample, time = sampleTime });
+    }
 
-  /// <summary> Returns the average change between each sample per unit time, or zero if the buffer is not full. </summary>
-  public abstract T Delta();
+    public void SetLatest(SampleType sample, float sampleTime) {
+      if (Count == 0) Set(0, sample, sampleTime);
+      else Set(Count - 1, sample, sampleTime);
+    }
 
-}
+    public float GetTime(int idx) {
+      return _buffer.Get(idx).time;
+    }
 
-/// <summary> Allows you to add to a capped-size ring buffer of Vector3s and,
-/// when full, compute the buffer's average change over time. </summary>
-public class DeltaBuffer : DeltaBuffer<Vector3> {
+    /// <summary>
+    /// Returns the average change between each sample per unit time.
+    /// 
+    /// If the buffer is empty, you should return the identity for your derivative type.
+    /// </summary>
+    public abstract DerivativeType Delta();
 
-  public DeltaBuffer(int bufferSize) : base(bufferSize) { }
+  }
 
-  /// <summary> Returns the average change between each sample per unit time, or zero if the buffer is not full. </summary>
-  public override Vector3 Delta() {
-    if (!IsFull) {
-      return Vector3.zero;
-    }
-    Vector3 deltaPerTimeSum = Vector3.zero;
-    int length = Length;
-    for (int i = 0; i < length - 1; i++) {
-      deltaPerTimeSum += (Get(i + 1) - Get(i)) / (GetTime(i + 1) - GetTime(i));
+  /// <summary>
+  /// A ring buffer of Vector3s with a Delta() function that computes the buffer's
+  /// average change over time.
+  /// 
+  /// The larger the buffer, the more stable but also delayed the resulting average
+  /// change over time. A buffer size of 5 is a good start for 60-90 Hz updates.
+  /// </summary>
+  public class DeltaBuffer : DeltaBuffer<Vector3, Vector3> {
+
+    public DeltaBuffer(int bufferSize) : base(bufferSize) { }
+
+    /// <summary>
+    /// Returns the average change between each sample per unit time, or zero if the
+    /// buffer is empty.
+    /// 
+    /// The larger the buffer, the more stable but also delayed the resulting average
+    /// change over time. A buffer size of 5 is a good start for 60-90 Hz updates.
+    /// </summary>
+    public override Vector3 Delta() {
+      if (Count == 0) { return Vector3.zero; }
+
+      Vector3 deltaPerTimeSum = Vector3.zero;
+      for (int i = 0; i < Count - 1; i++) {
+        deltaPerTimeSum += (Get(i + 1) - Get(i)) / (GetTime(i + 1) - GetTime(i));
+      }
+      return deltaPerTimeSum / Count;
```

Comment:
Is this intended that the logic changed from dividing by `length - 1`, to just dividing by `Count`?  As far as I can see Length was just renamed to Count, so why did this logic change?

Reply:
Ah, woops, good catch, I didn't look at this one closely enough. I need to be returning zero if the buffer is empty _or_ if the buffer contains only one sample, which will then allow me to properly do "Count - 1" to account for the fact that each delta change is a comparison of two consecutive samples. Will fix.